File: project/core/module/rest-service/src/main/java/vdi/service/rest/config/UploadConfig.java
package vdi.service.rest.config;

import vdi.core.config.vdi.RestServiceConfig;

public record UploadConfig(long maxUploadSize, long userMaxStorageSize) {
  private static final long DefaultMaxUploadSize = 1073741824L;
  private static final long DefaultMaxStorageSize = 10737418240L;

  public UploadConfig(RestServiceConfig config) {
    this(
      config == null ? DefaultMaxUploadSize : config.getLongMaxUploadSize(),
      config == null ? DefaultMaxStorageSize : config.getLongUserMaxStorageSize()
    );
  }
}

File: project/core/module/rest-service/src/main/kotlin/vdi/service/rest/config/UploadConfig.kt
package vdi.service.rest.config

import vdi.core.config.vdi.RestServiceConfig

data class UploadConfig(val maxUploadSize: ULong, val userMaxStorageSize: ULong) {
  constructor(config: RestServiceConfig?): this(
    maxUploadSize      = config?.maxUploadSize ?: MaxUploadSize,
    userMaxStorageSize = config?.userMaxStorageSize ?: MaxStorageSize,
  )

  private companion object {
    inline val MaxUploadSize get() = 1073741824uL
    inline val MaxStorageSize get() = 10737418240uL
  }
}
